tests/testthat/test-save-random-seed.R

```diff
@@ -0,0 +1,46 @@
+bc <- BatchContainer$new(
+  dimensions = c(location = 20)
+) |>
+  assign_in_order(
+    data.frame(
+      sampleId = seq_len(20)
+    )
+  )
+
+scoring_f <- function(...) rnorm(1)
+
+test_that("random seed is saved when no seed is set", {
+  # unset random seed
+  rm(.Random.seed, envir = .GlobalEnv)
```

Comment:
hi @klmr ,
in this file I'm testing that random seed and rng kind is saved in `bc$trace` both when:
- random seed is not set yet
- random seed is already set (that's easy)

I'm using `rm(.Random.seed, envir = .GlobalEnv)` to simulate a situation when random seed wasn't set


Do you think the reviewer would argue that this is bad code since it modifies the global environment?

Do you have a better suggestion how to test this?

- `set.seed(NULL)` doesn't seem to remove `.Random.seed`
- I couldn't find a way to start the test in a clean environment.
- if I use empty env, nothing works (since e.g. `{` is undefined)
- something with `local_mocked_bindings()`?